modules/instrument_list/php/instrument_list.class.inc

```diff
@@ -345,11 +370,11 @@ class Instrument_List extends \NDB_Menu_Filter
 
         if (!is_null($stage)) {
             $query .= "
-            	LEFT JOIN test_battery b 
-       	        ON ((t.Test_name=b.Test_name OR b.Test_name IS NULL) 
-      	        AND (s.SubprojectID=b.SubprojectID OR b.SubprojectID IS NULL) 
-       	        AND (s.Visit_label=b.Visit_label OR b.Visit_label IS NULL) 
-       	        AND (s.CenterID=b.CenterID OR b.CenterID IS NULL)) ";
+                LEFT JOIN test_battery b
```

Comment:
Shouldn't we also add a clause to check whether `b.Active='Y'`?

Reply:
I don't think this is necessary, to not have more performance penalty.

Reply:
If you do not add that check, inactive visits might be returned by this method, which is a bug. This check is definitely needed.

Reply:
Oh, test_battery active or not might be another issue, nothing to do with module active or not. I can create another PR if necessary

Reply:
You are right, it does have to be part of this PR but since it is such a small change, you might as well incorporate it here. Your call. Thanks!

Reply:
PR #6418 was submitted, I feel that Active = 'Y' check is not mandatory since it is in a left join, so I eventually did not add that check in this PR.